packages/sanity/src/structure/comments/plugin/field/CommentsField.tsx

```diff
@@ -152,6 +155,16 @@ function CommentFieldInner(props: FieldProps) {
       return
     }
 
+    if (commentsEnabled === 'read-only') {
+      if (upsellData) {
+        setUpsellDialogOpen(true)
+      } else {
+        // Open the comments inspector
```

Comment:
It will open the comment inspector if for some reason the upsell data hasn't loaded yet to avoid stale clicks on the comments icon. 
It should not allow to add new comments when it's read only.